x/wasm/keeper/events.go

```diff
@@ -50,8 +51,9 @@ func contractSDKEventAttributes(customAttributes []wasmvmtypes.EventAttribute, c
 	attrs := []sdk.Attribute{sdk.NewAttribute(types.AttributeKeyContractAddr, contractAddr.String())}
 	// append attributes from wasm to the sdk.Event
 	for _, l := range customAttributes {
-		// and reserve the contract_address key for our use (not contract)
-		if l.Key != types.AttributeKeyContractAddr {
+		// FIXME: do we want to error here on invalid events
+		// and reserve all _* keys for our use (not contract)
```

Comment:
We do!

Reply:
Same for the min length check in `newCustomEvents` above. Why silently ignore events with invalid type length?

Reply:
This was Alex's choice.
I left a message on the v0.15.0 upgrade PR to that effect. (See https://github.com/CosmWasm/wasmd/pull/548#discussion_r674203149 )

I would make that change in a separate PR as it is unrelated to v0.16.0. But good not to miss it.

Reply:
Could you create a ticket for those two places? Then we can merge here.